src/spanningtrees/boruvka.jl

```diff
@@ -0,0 +1,56 @@
+@traitfn function boruvka_mst(graph::AG::(!IsDirected), 
+    distmx::AbstractMatrix{T} = weights(graph); 
+    minimize = true) where {T<:Real,U,AG<:AbstractGraph{U}}
+    
+    f_distmx = minimize ? copy(distmx) : copy(-distmx)
+
+    djSet = IntDisjointSets(nv(graph))
+
+    mst = Vector{edgetype(graph)}()
+    sizehint!(mst, nv(graph) - 1)
+
+    weight = 0.0
+    while (num_groups(djSet) > 1)
+
+        cheapest = Array{Union{edgetype(graph), Bool}}(undef, nv(graph))
```

Comment:
So an element of this Array is either an edge or the value `false`? In that case I would suggest to use `Array{Union{edgetype(graph), Nothing}}` instead. And maybe use `Vector` instead of `Array` -- the result is the same, but `Vector` is easier to understand.

Reply:
alternatively, you could make a sentinel value of `Edge(0, 0)` (since that will never appear in any graph edgelist), and then not worry about a union.

Reply:
> So an element of this Array is either an edge or the value `false`? In that case I would suggest to use `Array{Union{edgetype(graph), Nothing}}` instead. And maybe use `Vector` instead of `Array` -- the result is the same, but `Vector` is easier to understand.

What would be the difference between using " false " and " nothing " ?

Reply:
Such a sentinel edge might have the problem that it only works for `SimpleGraph` and not other types.

The difference between `false` and `nothing` is their type, `Boolean` and `Nothing` respectively. `Boolean` has two values, `true` and `false`, while `Nothing` can only have the value `nothing`.

As far as I know, Julia has some special optimizations for `Vector{Union{X, Nothing}}` - it stores the information on wether an element of this vector is an `X` or a `Nothing` in a separate bit vector. So basically it uses less space.

Reply:
> Such a sentinel edge might have the problem that it only works for `SimpleGraph` and not other types.
> 
> The difference between `false` and `nothing` is their type, `Boolean` and `Nothing` respectively. `Boolean` has two values, `true` and `false`, while `Nothing` can only have the value `nothing`.
> 
> As far as I know, Julia has some special optimizations for `Vector{Union{X, Nothing}}` - it stores the information on wether an element of this vector is an `X` or a `Nothing` in a separate bit vector. So basically it uses less space.

oh thanks! Got it, changing it